core/src/datayoga_core/blocks/relational/utils.py

```diff
@@ -39,15 +40,29 @@ def get_engine(connection_name: str, context: Context, autocommit: bool = True)
         lib_dir = connection.get("oracle_thick_mode_lib_dir")
         extra["thick_mode"] = {'lib_dir': lib_dir} if lib_dir else {}
 
+    # PSQL specific parameters to be passed in the query string for SSL connection
+    if db_type == DbType.PSQL:
+        args = connection.get("connect_args", {})
+        ssl_args["sslmode"] = args.get("sslmode")
+        ssl_args["sslrootcert"] = args.get("sslrootcert")
+        ssl_args["sslkey"] = args.get("sslkey")
+        ssl_args["sslcert"] = args.get("sslcert")
+        ssl_args["sslpassword"] = args.get("sslpassword")
+
+        connection["connect_args"] = {k: v for k, v in args.items() if k not in ssl_args.keys()}
+        ssl_args = {k: v for k, v in ssl_args.items() if v is not None}
+
     engine = sa.create_engine(
         sa.engine.URL.create(
             drivername=connection.get("driver", DEFAULT_DRIVERS.get(db_type)),
             host=connection.get("host"),
             port=connection.get("port"),
             username=connection.get("user"),
             password=connection.get("password"),
-            database=connection.get("database")),
-        echo=connection.get("debug", False), connect_args=connection.get("connect_args", {}),
+            database=connection.get("database"),
+            query=ssl_args),
```

Comment:
shouldn't we just pass these in as "query"? I guess there will be others that are part of this query, not just SSL

Reply:
✅